go/chat/boxer.go

```diff
@@ -1100,7 +1117,29 @@ func (b *Boxer) BoxMessage(ctx context.Context, msg chat1.MessagePlaintext,
 		return nil, NewBoxingError(msg, false)
 	}
 
-	err = b.attachMerkleRoot(ctx, &msg)
+	// If the message is exploding, load the ephemeral key, and tweak the
+	// version. Make sure we're not using MessageBoxedVersion_V1, since that
+	// doesn't support exploding messages.
+	var ephemeralSeed *keybase1.TeamEk
+	if msg.IsExploding() {
+		ek, err := CtxKeyFinder(ctx, b.G()).EphemeralKeyForEncryption(
+			ctx, tlfName, msg.ClientHeader.Conv.Tlfid, membersType, msg.ClientHeader.TlfPublic)
+		if err != nil {
+			return nil, NewBoxingCryptKeysError(err)
+		}
+		ephemeralSeed = &ek
+		if version == chat1.MessageBoxedVersion_V1 {
```

Comment:
can you move this up as a first check? we can just short circuit the keygen machinery if so